src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,66 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
+    return new ArrayLikeObservable(arrayLike, mapFn, thisArg, scheduler);
```

Comment:
If there's no `mapFn`, and `length` of the `arrayLike` is 1, you could return `ScalarObservable`, and if it's zero, `EmptyObservable`